### PR TITLE
feat(select): sync with @carbon/react

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[select]: sync with @carbon/react version #10116

### Description

@carbon/react syncing with `select` component. 

@carbon/react version: https://react.carbondesignsystem.com/?path=/story/components-select--playground

Also adjusted the recently sync'ed `layer` component styles to match v11.

### Changelog

**Changed**

- select update to match v11 storybook
- move slotted styles for `layer` - they are meant for the storybook example instead

### CODE REVIEW

**Review by kennylam — DISMISSED** [review]
@annawen1 getting this build error for both dotcom web components and react wrappers:

image

Probably the prefix change causing the issue in another file? Ordinarily we could just fix later, but since it breaks, deploy previews get canceled in Jenkins.

**Comment by RichKummer** [comment]
@annawen1 `select` is working great. I noticed a 1px height difference between react and WC – I'm guessing it's the bottom border/rule?

**React:**
Screen Shot 2023-03-30 at 5 22 53 PM

**WC:**
Screen Shot 2023-03-30 at 5 22 40 PM

File: packages/carbon-web-components/src/components/select/select-skeleton.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * `true` if the label should be hidden. Corresponds to the attribute with the same name.
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'hide-label' })
+  hideLabel = false;
+
+  render() {
+    const { hideLabel } = this;
+    return html`
+      ${!hideLabel &&
```
[the remark this answers is not in the document]
Comment (kennylam) [reply packages/carbon-web-components/src/components/select/select-skeleton.ts:29]: Oh ok, I didn't see a knob in the story so got confused.

**Review by kennylam — APPROVED** [review]
LGTM!

**Comment by annawen1** [comment]
> @annawen1 `select` is working great. I noticed a 1px height difference between react and WC – I'm guessing it's the bottom border/rule?
> 
> **React:** Screen Shot 2023-03-30 at 5 22 53 PM
> 
> **WC:** Screen Shot 2023-03-30 at 5 22 40 PM

weird, so the styles are the exact same coming from carbon core so I'm not too sure what could be causing the height to differ by 1px. Both have the border 1px applied via styles so it should be the same - my guess is that because the sizes are `rem` based and we still have to fix the plex styles being imported - it could be causing this small difference. I'm going to merge it in for now and check back again when we fix the font import issue.

**Review by RichKummer — APPROVED** [review]

**Pull request merged**